src/transformers/models/fast/image_processing_fast.py

```diff
@@ -0,0 +1,615 @@
+        )  # B*1*320*320
+        texts = self._max_pooling(texts, scale=scale)  # B*1*320*320
+        score_maps = torch.sigmoid_(texts)  # B*1*320*320
+        score_maps = F.interpolate(score_maps, size=(image_size[0], image_size[1]), mode="nearest")  # B*1*640*640
+        score_maps = score_maps.squeeze(1)  # B*640*640
+
+        kernels = (out[:, 0, :, :] > 0).to(torch.uint8)  # B*160*160
+        labels_ = []
+        for kernel in kernels.numpy():
+            ret, label_ = connected_components(kernel)
+            labels_.append(label_)
+        labels_ = np.array(labels_)
+        labels_ = torch.from_numpy(labels_)
+        labels = labels_.unsqueeze(1).to(torch.float32)  # B*1*160*160
+        labels = F.interpolate(
+            labels, size=(image_size[0] // scale, image_size[1] // scale), mode="nearest"
+        )  # B*1*320*320
+        labels = self._max_pooling(labels, scale=scale)
+        labels = F.interpolate(labels, size=(image_size[0], image_size[1]), mode="nearest")  # B*1*640*640
+        labels = labels.squeeze(1).to(torch.int32)  # B*640*640
+
+        keys = [torch.unique(labels_[i], sorted=True) for i in range(batch_size)]
+
+        final_results.update({"kernels": kernels.data.cpu()})
+
+        results = []
+        for i in range(batch_size):
+            org_image_size = target_sizes[i]
+            scales = (float(org_image_size[1]) / float(image_size[1]), float(org_image_size[0]) / float(image_size[0]))
+
+            bounding_boxes, scores = self.generate_bounding_box(
+                keys[i], labels[i], score_maps[i], scales, threshold, bounding_box_type=bounding_box_type
+            )
+            results.append({"boxes": bounding_boxes, "scores": scores})
+        final_results.update({"results": results})
+
+        return results
+
+    def generate_bounding_box(self, keys, label, score, scales, threshold, bounding_box_type):
+        """
+        Generates bounding boxes and corresponding confidence scores from instance labels and score maps.
+
+        Args:
+            keys (Tensor): Unique instance labels (1D Tensor) for connected components in the image.
+            label (Tensor): Instance segmentation map (H x W), where each connected region has a unique label.
+            score (Tensor): Confidence map (H x W) with values in [0, 1], representing text region probabilities.
+            scales (Tuple[float, float]): Scaling factors (width_scale, height_scale) used to map results
+                                        back to original image dimensions.
+            threshold (float): Minimum average score required for a region to be considered a valid detection.
+            bounding_box_type (str): Type of bounding box to generate for each instance. Options:
+                            - "rect": Minimum area rotated rectangle (via `compute_min_area_rect`).
+                            - "poly": Polygonal contour (via `cv2.findContours`, requires OpenCV).
+
```

Comment:
I suggest to split this into to separate methods, because they return different outputs, one for polygons, another one for bounding boxes